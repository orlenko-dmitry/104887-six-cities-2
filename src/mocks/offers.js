const offers = [{
  "id": 1,
  "city": {
    "name": `Amsterdam`,
    "location": {
      "latitude": 52.370216,
      "longitude": 4.895168,
      "zoom": 10
    }
  },
  "previewImage": `img/apartment-01.jpg`,
  "images": [`/img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`],
  "title": `Beautiful & luxurious studio at great location`,
  "isFavorite": true,
  "isPremium": true,
  "rating": 4.8,
  "type": `apartment`,
  "bedrooms": 3,
  "max_adults": 4,
  "price": 120,
  "goods": [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  "host": {
    "id": 3,
    "isPro": true,
    "name": `Angelina`,
    "avatar_url": `img/avatar-angelina.jpg`
  },
  "description": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  "location": {
    "latitude": 52.35514938496378,
    "longitude": 4.673877537499948,
    "zoom": 8
  }
}, {
  "id": 2,
  "city": {
    "name": `Amsterdam`,
    "location": {
      "latitude": 52.370216,
      "longitude": 4.895168,
      "zoom": 10
    }
  },
  "previewImage": `img/room.jpg`,
  "images": [`/img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`],
  "title": `Wood and stone place`,
  "isFavorite": false,
  "isPremium": false,
  "rating": 4.2,
  "type": `private room`,
  "bedrooms": 1,
  "max_adults": 2,
  "price": 80,
  "goods": [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  "host": {
    "id": 3,
    "isPro": true,
    "name": `Angelina`,
    "avatar_url": `img/avatar-angelina.jpg`
  },
  "description": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  "location": {
    "latitude": 52.35514938496378,
    "longitude": 4.673877537499948,
    "zoom": 8
  }
}, {
  "id": 3,
  "city": {
    "name": `Amsterdam`,
    "location": {
      "latitude": 52.370216,
      "longitude": 4.895168,
      "zoom": 10
    }
  },
  "previewImage": `img/apartment-02.jpg`,
  "images": [`/img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`],
  "title": `Nice, cozy, warm big bed apartmen`,
  "isFavorite": true,
  "isPremium": false,
  "rating": 4,
  "type": `apartment`,
  "bedrooms": 2,
  "max_adults": 3,
  "price": 70,
  "goods": [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  "host": {
    "id": 3,
    "isPro": true,
    "name": `Angelina`,
    "avatar_url": `img/avatar-angelina.jpg`
  },
  "description": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  "location": {
    "latitude": 52.35514938496378,
    "longitude": 4.673877537499948,
    "zoom": 8
  }
}, {
  "id": 4,
  "city": {
    "name": `Amsterdam`,
    "location": {
      "latitude": 52.370216,
      "longitude": 4.895168,
      "zoom": 10
    }
  },
  "previewImage": `img/apartment-03.jpg`,
  "images": [`/img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`],
  "title": `Beautiful luxurious apartment at great location`,
  "isFavorite": false,
  "isPremium": true,
  "rating": 4.9,
  "type": `apartment`,
  "bedrooms": 3,
  "max_adults": 6,
  "price": 200,
  "goods": [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  "host": {
    "id": 3,
    "isPro": true,
    "name": `Angelina`,
    "avatar_url": `img/avatar-angelina.jpg`
  },
  "description": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  "location": {
    "latitude": 52.35514938496378,
    "longitude": 4.673877537499948,
    "zoom": 8
  }
}];

export default offers;
