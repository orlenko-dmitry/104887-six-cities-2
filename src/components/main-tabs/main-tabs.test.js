import React from 'react';
import {shallow} from 'enzyme';

import MainTabs from '../main-tabs/main-tabs.jsx';
import offers from '../../mocks/offers.js';
import {APP_CITIES} from '../../consts/index.js';

const city = offers[0].city;

it(`MainTabs rnders correctly`, () => {
  const tree = shallow(
      <MainTabs
        cities={APP_CITIES}
        selectedCity={city}
        onSelectCityClick={() => {}}
      />
  );

  expect(tree).toMatchSnapshot();
});
