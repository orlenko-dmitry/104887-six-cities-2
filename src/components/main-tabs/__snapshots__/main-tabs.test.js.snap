// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainTabs rnders correctly 1`] = `
<div
  className="tabs"
>
  <section
    className="locations container"
  >
    <ul
      className="locations__list tabs__list"
    >
      <li
        className="locations__item"
        data-testid="city-0"
        key="Amsterdam"
        onClick={[Function]}
      >
        <a
          className="locations__item-link tabs__item tabs__item--active"
        >
          <span>
            Amsterdam
          </span>
        </a>
      </li>
      <li
        className="locations__item"
        data-testid="city-1"
        key="Brussels"
        onClick={[Function]}
      >
        <a
          className="locations__item-link tabs__item "
        >
          <span>
            Brussels
          </span>
        </a>
      </li>
      <li
        className="locations__item"
        data-testid="city-2"
        key="Cologne"
        onClick={[Function]}
      >
        <a
          className="locations__item-link tabs__item "
        >
          <span>
            Cologne
          </span>
        </a>
      </li>
      <li
        className="locations__item"
        data-testid="city-3"
        key="Paris"
        onClick={[Function]}
      >
        <a
          className="locations__item-link tabs__item "
        >
          <span>
            Paris
          </span>
        </a>
      </li>
      <li
        className="locations__item"
        data-testid="city-4"
        key="Hamburg"
        onClick={[Function]}
      >
        <a
          className="locations__item-link tabs__item "
        >
          <span>
            Hamburg
          </span>
        </a>
      </li>
      <li
        className="locations__item"
        data-testid="city-5"
        key="Dusseldorf"
        onClick={[Function]}
      >
        <a
          className="locations__item-link tabs__item "
        >
          <span>
            Dusseldorf
          </span>
        </a>
      </li>
    </ul>
  </section>
</div>
`;
